[CI] ChunkedZipResponseIT testGetNextPartFailure failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3684 / rhel-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/dcnlw2bq5t6hu)
- [elasticsearch-pull-request #27884 / part-1](https://gradle-enterprise.elastic.co/s/fi7hutwupfpwi)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.rest.ChunkedZipResponseIT.testGetNextPartFailure" -Dtests.seed=81E99CF38948552C -Dtests.locale=mt -Dtests.timezone=America/Atikokan -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testGetNextPartFailure'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.rest.ChunkedZipResponseIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [main] 2 failures in test testGetNextPartFailure (2.2% fail rate in 93 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -135,9 +135,6 @@ tests:
   method: testMatchCommand
   issue: https://github.com/elastic/elasticsearch/issues/111661
 - class: org.elasticsearch.rest.ChunkedZipResponseIT
-  method: testGetNextPartFailure
-  issue: https://github.com/elastic/elasticsearch/issues/111697
-- class: org.elasticsearch.rest.ChunkedZipResponseIT
   method: testAbort
   issue: https://github.com/elastic/elasticsearch/issues/111698
 

--- a/server/src/internalClusterTest/java/org/elasticsearch/rest/ChunkedZipResponseIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/rest/ChunkedZipResponseIT.java
@@ -163,7 +163,7 @@ public class ChunkedZipResponseIT extends ESIntegTestCase {
                     final var getNextPartCountDown = request.paramAsInt(GET_NEXT_PART_COUNT_DOWN_PARAM, -1);
                     final Runnable onGetNextPart;
                     final Supplier<EntryBody> entryBodySupplier;
-                    if (getNextPartCountDown <= 1) {
+                    if (getNextPartCountDown <= 0) {
                         onGetNextPart = () -> {};
                         entryBodySupplier = () -> randomContent(between(1, 10), ByteSizeUnit.MB.toIntBytes(1));
                     } else {

